cli/src/stake.rs

```diff
@@ -0,0 +1,742 @@
+        }
+        Err(err) => Err(WalletError::RpcRequestError(format!(
+            "Account data could not be deserialized to stake state: {:?}",
+            err
+        )))?,
+    }
+}
+
+pub fn process_delegate_stake(
+    rpc_client: &RpcClient,
+    config: &WalletConfig,
+    stake_account_pubkey: &Pubkey,
+    vote_account_pubkey: &Pubkey,
+    force: bool,
+) -> ProcessResult {
+    check_unique_pubkeys(
+        (&config.keypair.pubkey(), "wallet keypair".to_string()),
+        (stake_account_pubkey, "stake_account_pubkey".to_string()),
+    )?;
+
+    // Sanity check the vote account to ensure it is attached to a validator that has recently
+    // voted at the tip of the ledger
+    let vote_account_data = rpc_client
+        .get_account_data(vote_account_pubkey)
+        .map_err(|_| {
+            WalletError::RpcRequestError(format!("Vote account not found: {}", vote_account_pubkey))
+        })?;
+
+    let vote_state = VoteState::deserialize(&vote_account_data).map_err(|_| {
+        WalletError::RpcRequestError(
+            "Account data could not be deserialized to vote state".to_string(),
+        )
+    })?;
+
+    let sanity_check_result = match vote_state.root_slot {
+        None => Err(WalletError::BadParameter(
+            "Unable to delegate. Vote account has no root slot".to_string(),
+        )),
+        Some(root_slot) => {
+            let slot = rpc_client.get_slot()?;
+            if root_slot + solana_sdk::clock::DEFAULT_SLOTS_PER_TURN < slot {
+                Err(WalletError::BadParameter(
+                    format!(
+                    "Unable to delegate. Vote account root slot ({}) is too old, the current slot is {}", root_slot, slot
+                    )
+                ))
+            } else {
+                Ok(())
+            }
+        }
+    };
+
+    if sanity_check_result.is_err() {
+        if !force {
+            sanity_check_result?;
+        } else {
+            println!("--force supplied, ignoring: {:?}", sanity_check_result);
+        }
+    }
+
+    let (recent_blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+
+    let ixs = vec![stake_instruction::delegate_stake(
+        stake_account_pubkey,
+        &config.keypair.pubkey(),
+        vote_account_pubkey,
+    )];
+
+    let mut tx = Transaction::new_signed_instructions(&[&config.keypair], ixs, recent_blockhash);
+    check_account_for_fee(rpc_client, config, &fee_calculator, &tx.message)?;
+    let result = rpc_client.send_and_confirm_transaction(&mut tx, &[&config.keypair]);
+    log_instruction_custom_error::<StakeError>(result)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::wallet::{app, parse_command};
+
+    #[test]
+    fn test_parse_command() {
+        let test_commands = app("test", "desc", "version");
+        let pubkey = Pubkey::new_rand();
+        let pubkey_string = format!("{}", pubkey);
+
+        //        // Test AuthorizeStaker Subcommand
```

Comment:
Should this section be uncommented? Or removed?

Reply:
uncommented